src/app/digital/views/components/NANDGateView.ts

```diff
@@ -0,0 +1,55 @@
+import {DEFAULT_BORDER_COLOR, DEFAULT_BORDER_WIDTH, DEFAULT_CURVE_BORDER_WIDTH, SELECTED_BORDER_COLOR} from "core/utils/Constants";
+
+import {V} from "Vector";
+
+import {Rect} from "math/Rect";
+
+import {Style} from "core/utils/rendering/Style";
+
+import {Line} from "core/utils/rendering/shapes/Line";
+
+import {NANDGate} from "core/models/types/digital";
+
+import {RenderInfo}    from "core/views/BaseView";
+import {ComponentView} from "core/views/ComponentView";
+
+import {DigitalViewInfo} from "../DigitalViewInfo";
+
+
+export class NANDGateView extends ComponentView<NANDGate, DigitalViewInfo> {
```

Comment:
Notice how similar this is to `ANDGateView`, we probably want to re-use `ANDGateView`, so to do this, we can have `ANDGateView` extend `ComponentView<ANDGate | NANDGate` and take in an `ANDGate | NANDGate` and then re-use this in `src/app/digital/views/index.ts`